src-java/ping-topology/ping-storm-topology/src/test/java/org/openkilda/wfm/topology/ping/PingTopologyTest.java

```diff
@@ -0,0 +1,499 @@
+    private static HaSubFlowRepository haSubFlowRepository;
+    private static SwitchRepository switchRepository;
+
+    private Switch switch1;
+    private Switch switch2;
+    private Switch switch3;
+    private Switch switch4;
+
+    @BeforeClass
+    public static void setupOnce() throws Exception {
+        Properties configOverlay = getZooKeeperProperties(FLOW_PING_TOPOLOGY_TEST_ZOOKEEPER_PORT, ROOT_NODE);
+        configOverlay.putAll(getKafkaProperties(FLOW_PING_TOPOLOGY_TEST_KAFKA_PORT));
+
+        AbstractStormTest.startZooKafka(configOverlay);
+        setStartSignal(FLOW_PING_TOPOLOGY_TEST_ZOOKEEPER_PORT, ROOT_NODE, COMPONENT_NAME, RUN_ID);
+        AbstractStormTest.startStorm(COMPONENT_NAME, RUN_ID);
+
+        LaunchEnvironment launchEnvironment = makeLaunchEnvironment();
+        launchEnvironment.setupOverlay(configOverlay);
+
+        persistenceManager = InMemoryGraphPersistenceManager.newInstance();
+        persistenceManager.install();
+
+        PingTopology pingTopology = new PingTopology(launchEnvironment, persistenceManager);
+        pingTopologyConfig = pingTopology.getConfig();
+        StormTopology stormTopology = pingTopology.createTopology();
+        Config config = stormConfig();
+        cluster.submitTopology(PingTopologyTest.class.getSimpleName(), config, stormTopology);
+
+
+        speakerConsumer = new TestKafkaConsumer(pingTopologyConfig.getKafkaSpeakerFlowPingTopic(),
+                kafkaConsumerProperties(UUID.randomUUID().toString(), "floodlight", RUN_ID), 100);
+        speakerConsumer.start();
+
+        northboundConsumer = new TestKafkaConsumer(pingTopologyConfig.getKafkaNorthboundTopic(),
+                kafkaConsumerProperties(UUID.randomUUID().toString(), "northbound", RUN_ID), 10000);
+        northboundConsumer.start();
+
+        haFlowPathRepository = persistenceManager.getRepositoryFactory().createHaFlowPathRepository();
+        flowPathRepository = persistenceManager.getRepositoryFactory().createFlowPathRepository();
+        haFlowRepository = persistenceManager.getRepositoryFactory().createHaFlowRepository();
+        haSubFlowRepository = persistenceManager.getRepositoryFactory().createHaSubFlowRepository();
+        switchRepository = persistenceManager.getRepositoryFactory().createSwitchRepository();
+        transitVlanRepository = persistenceManager.getRepositoryFactory().createTransitVlanRepository();
+
+        sleep(TOPOLOGY_START_TIMEOUT);
+    }
+
+    @AfterClass
+    public static void teardownOnce() throws Exception {
+        speakerConsumer.wakeup();
+        speakerConsumer.join();
+        northboundConsumer.wakeup();
+        northboundConsumer.join();
+        AbstractStormTest.stopZooKafkaAndStorm();
+    }
+
+    @After
+    public void teardown() {
+        persistenceManager.getInMemoryImplementation().purgeData();
+        speakerConsumer.clear();
+        northboundConsumer.clear();
+        sleep(500);
```

Comment:
I think you need to add an assertion after waking up that everything is clean up. Otherwise it is more like fingers crossed.
If the test would fail because of clean up issues, we'll know about it and can adjust the timing. Otherwise the next test in the queue might be flaky and we don't know why.